knative/operator
# CR field for default channel

Fixes #23

## Proposed Changes

* New CR field for default channel template
* Defaults to `InMemoryChannel`
* Did some refactoring in brokerclass defaults code to share code with channel defaults

**Release Note**



```release-note
It is now possible to set the default channel template within KnativeEventing CR with `defaultChannelTemplate` field. Defaults to `messaging.knative.dev/v1beta InMemoryChannel`.
```

### Test instructions
Apply following:
```yaml
apiVersion: operator.knative.dev/v1alpha1
kind: KnativeEventing
metadata:
  name: knative-eventing
  namespace: knative-eventing
spec:
  defaultChannelTemplate:
    apiVersion: messaging.knative.dev/v1alpha1
    kind: KafkaChannel
    spec:
      numPartitions: 10
      replicationFactor: 1
```
Check `default-ch-webhook` configmap.
You should see
```
data:
  default-ch-config: |
    clusterDefault:
      apiVersion: messaging.knative.dev/v1alpha1
      kind: KafkaChannel
```
